transport/src/main/java/io/netty/channel/AbstractChannelHandlerContext.java

```diff
@@ -833,6 +833,14 @@ private void write(Object msg, boolean flush, ChannelPromise promise) {
                 task = WriteTask.newInstance(next, m, promise);
             }
             safeExecute(executor, task, promise, m);
+            if (!flush) {
+                /**
+                 * Whenever there is a write, if auto-flush is enabled, the eventloop is required to be woken up,
+                 * otherwise the write may never be flushed or have latency.
+                 * This method just wakes up the eventloop once till the next auto-flush task is run.
+                 */
+                pipeline.wakeUpForAutoFlushIfRequired();
```

Comment:
Since a handler is invoked by the eventloop, there is no use of waking up the eventloop inside the handler.